modules/browser-node/src/index.ts

```diff
@@ -15,94 +15,253 @@ import {
+    supportedChains?: number[];
+    iframeSrc?: string;
+  }) {
+    this.logger = params.logger || pino();
+    this.routerPublicIdentifier = params.routerPublicIdentifier;
+    this.supportedChains = params.supportedChains || [];
+    this.iframeSrc = params.iframeSrc;
+  }
+
+  // method for non-signer based apps to connect to iframe
+  async init(): Promise<void> {
+    // TODO: validate config
+    let iframeSrc = this.iframeSrc;
+    if (!iframeSrc) {
+      iframeSrc = "https://wallet.connext.network";
+    }
+    this.logger.info({ method: "connect", iframeSrc }, "Connecting with iframe provider");
+    this.channelProvider = await IframeChannelProvider.connect({
+      src: iframeSrc!,
+      id: "connext-iframe",
+    });
+    const rpc = constructRpcRequest("connext_authenticate", {});
+    const auth = await this.channelProvider.send(rpc);
+    this.logger.info({ method: "connect", response: auth }, "Received response from auth method");
+    const [nodeConfig] = await this.getConfig();
+    this.publicIdentifier = nodeConfig.publicIdentifier;
+    this.signerAddress = nodeConfig.signerAddress;
+    this.logger.info(
+      { supportedChains: this.supportedChains, routerPublicIdentifier: this.routerPublicIdentifier },
+      "Checking for existing channels",
+    );
+    for (const chainId of this.supportedChains) {
+      let channel = await this.getStateChannelByParticipants({
+        chainId,
+        counterparty: this.routerPublicIdentifier!,
       });
-      await messaging.connect();
-      const store = new BrowserStore(config.logger.child({ module: "BrowserStore" }));
-      const lock = new BrowserLockService(
-        config.signer.publicIdentifier,
-        messaging,
-        config.logger.child({ module: "BrowserLockService" }),
-      );
-      const chainService = new VectorChainService(
-        store,
-        chainJsonProviders,
-        config.signer,
-        config.logger.child({ module: "VectorChainService" }),
-      );
-      const engine = await VectorEngine.connect(
-        messaging,
-        lock,
-        store,
-        config.signer,
-        chainService,
-        config.chainAddresses!,
-        config.logger.child({ module: "VectorEngine" }),
+      if (!channel) {
+        this.logger.info({ chainId }, "Setting up channel");
+        const address = await this.setup({
+          chainId,
+          counterpartyIdentifier: this.routerPublicIdentifier!,
+          timeout: "100000",
+        });
+        if (address.isError) {
+          throw address.getError();
+        }
+        channel = await this.getStateChannel(address.getValue());
+      }
+      this.logger.info({ channel, chainId });
+    }
+  }
+
+  // IFRAME SPECIFIC
+  async crossChainTransfer(params: {
+    amount: string;
+    fromChainId: number;
+    fromAssetId: string;
+    toChainId: number;
+    toAssetId: string;
+    reconcileDeposit?: boolean;
+    withdrawalAddress?: string;
+  }): Promise<void> {
```

Comment:
Is this interface documented anywhere? Confused why we need to have `reconcileDeposit` or `withdrawalAddress` in here